### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Skye1510**
Code just now needs reviewing before merging

### CODE REVIEW

**Review by Regulr — COMMENTED** [review]
All Requirements met and code format in line with the Style Guidelines 

Code is well commmented and easy to read and understand 

No user input so little need for error handling but Debug.Assert used correctly to ientify logic errors

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -2,22 +2,19 @@
...
+namespace CMP1903_A1_2324 {
+  internal class Die {
+    private int _dieValue; //Private attribute
+    public int DieValue { 
+        get => _dieValue; set => _dieValue = value; //Getter and setter methods for encapsulation
+    }
+    public int Roll() {
+        var rand = new Random(); //create new random object
+        Thread.Sleep(100); //Pause for a moment to ensure new random numbers are generated
+        return rand.Next(1,7); //return an integer from between 1 and 6
```
Comment (Joshdunny) [inline CMP1903_A1_2324/Die.cs:17]: All requirements have been met for the Die class from what i can see!

Great formatting towards the style guide here with a good use of attributes

Die class is very readable with great comments 

Error of Random generating the same value based on seed has been handled correctly

File: CMP1903_A1_2324/Game.cs [hunk 2]
```diff
@@ -4,18 +4,21 @@
...
+      Die die; //Create a die variable
 
+      //Rolling the die 3 times
+      for (int i = 0; i < 3; i++) {
+        die = new Die(); //Create a new die object
+        die.DieValue = die.Roll(); //Roll the die, assign the result to the DieValue property
+        dieResults.Add(die.DieValue); //Add the die value to the list
+      }
+      
+      return dieResults; //Return the list
```
Comment (Joshdunny) [inline CMP1903_A1_2324/Game.cs:21]: Mandatory requirements have been met

Optional requirements such as reporting some stats on the rolls and allowing for multiple rolls to happen needs to be added for a first, which i recommend adding as think could be a first piece of code.

Formatting is on the ball to the style guide

readability is done well again with no chance of errors

File: CMP1903_A1_2324/Program.cs [hunk 3]
```diff
@@ -4,16 +4,21 @@
...
+      testing.Test();
+
+      Game game = new Game();
+
+      List<int> tempList = game.RollDie();
+
+      for (int i = 0; i < tempList.Count; i++){
+        Console.WriteLine("Roll " + (i+1) + " is " + tempList[i]);
+      }
+      Console.WriteLine("Sum of the rolls is: " + tempList.Sum());
```
Comment (Joshdunny) [inline CMP1903_A1_2324/Program.cs:21]: Great way of displaying the results of the rolls using the list collection

File: CMP1903_A1_2324/Program.cs [hunk 4]
```diff
@@ -4,16 +4,21 @@
...
     }
```
Comment (Joshdunny) [inline CMP1903_A1_2324/Program.cs:22]: All requirements are met for the program class 

Formatting is following the guide to the T

No commenting however, even though code is readable without

No errors occuring

File: CMP1903_A1_2324/Testing.cs [hunk 5]
```diff
@@ -1,20 +1,25 @@
...
+      Game game = new Game(); //Create new game object
+      List<int> tempList = game.RollDie(); //Make a temp list and assign its value to the return of the game object RollDie method
+      int sum = tempList.Sum(); //Make an integer variable, assign its value to be the sum of the items in the list
+
+      Debug.Assert(sum < 19 && sum > 2, "Sum is outside expected range"); //Check to ensure the sum is in the range 3 <= x <= 18
+    
+      //Check to ensure all rolls are within the valid range
+      for (int i = 0; i < tempList.Count; i++) {
+        Debug.Assert(tempList[i] > 0 && tempList[i] < 7, "Roll is outside of expected range");
+      }
```
Comment (Joshdunny) [inline CMP1903_A1_2324/Testing.cs:22]: All elements are added nicely but i have noticed for the testing class it is asked to create a new Die object and call its methods using the die object instead of using the game class. However you are still testing both classes correctly either way

Formatting is consistent again to the style guide

commenting allows for very readable code

**Review by Joshdunny — COMMENTED** [review]
Having An overlook at your code and testing it myself for any errors I can say this code is very well built, runs very well and nicely uses some methods of OOP 

From what i can tell all mandatory requirements are added, optional requirements however have not been implemented

Formatting of the code is consistent and follows the style guide very nicely!

Commenting is done through 3/4 with program not having any commenting

No errors occur, with errors such as Random generating the same value being handled well

In conclusion well done this is a great piece of code!
 I've also reviewed the code step by step with comments about each class including this summary.
